verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -155,26 +192,56 @@ void IndexingFactsTreeExtractor::ExtractModuleEnd(
 
   if (module_name != nullptr) {
     const Anchor module_end_anchor(*module_name, context_.base);
-    facts_tree_context_.back()->Value().AppendAnchor(module_end_anchor);
+    facts_tree_context_.top().Value().AppendAnchor(module_end_anchor);
   }
-}
+};
 
 void IndexingFactsTreeExtractor::ExtractModuleInstantiation(
     const verible::SyntaxTreeNode& node) {
+  std::vector<verible::TreeSearchMatch> gate_instances =
+      GetListOfGateInstanceFromDataDeclaration(node);
+
   const verible::TokenInfo& type =
       GetTypeTokenInfoFromModuleInstantiation(node);
   const Anchor type_anchor(type, context_.base);
 
-  const verible::TokenInfo& variable_name =
-      GetModuleInstanceNameTokenInfoFromDataDeclaration(node);
-  const Anchor variable_name_anchor(variable_name, context_.base);
+  for (const verible::TreeSearchMatch& instance : gate_instances) {
```

Comment:
suggest some // high-level comments interspersed, e.g. here

// Module instantiations (data declarations) may declare multiple instances sharing the same type in a single statement.
// Loop through each instance and associate each declared id with the same type.